bindinfo/session_handle.go

```diff
@@ -27,22 +27,23 @@ import (
 
 // SessionHandle is used to handle all session sql bind operations.
 type SessionHandle struct {
-	ch     cache
+	ch     *bindCache
 	parser *parser.Parser
 }
 
 // NewSessionBindHandle creates a new SessionBindHandle.
-func NewSessionBindHandle(parser *parser.Parser) *SessionHandle {
+func NewSessionBindHandle(ctx sessionctx.Context, parser *parser.Parser) *SessionHandle {
 	sessionHandle := &SessionHandle{parser: parser}
-	sessionHandle.ch = make(cache)
+	memCapacity := ctx.GetSessionVars().MemQuotaBindCache
+	sessionHandle.ch = newBindCache(memCapacity)
 	return sessionHandle
 }
 
 // appendBindRecord adds the BindRecord to the cache, all the stale bindMetas are
 // removed from the cache after this operation.
 func (h *SessionHandle) appendBindRecord(hash string, meta *BindRecord) {
```

Comment:
The naming here seems not consistent with similar logic in `handle.go`. I think it should be named `setBindRecord`.

Reply:
I will refactor this part in the later PR.